### PR TITLE
add scorecard.yml

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
@AlvaroBrey shall we then also enabled this for this repo?
If so, we would need to change sync.yml and also add this repo?

### CODE REVIEW

File: config/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
```
Comment (AlvaroBrey) [inline config/workflows/scorecard.yml:4]:
```suggestion
# synced from @nextcloud/android-config
```

File: config/workflows/scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          # repo_token: ${{ secrets.SCORECARD_TOKEN }}
+
+          # Public repositories:
+          #   - Publish results to OpenSSF REST API for easy access by consumers
+          #   - Allows the repository to include the Scorecard badge.
+          #   - See https://github.com/ossf/scorecard-action#publishing-results.
+          # For private repositories:
+          #   - `publish_results` will always be set to `false`, regardless
+          #     of the value entered here.
+          publish_results: true
```
Comment (AlvaroBrey) [inline config/workflows/scorecard.yml:57]:
```suggestion
          publish_results: false
```
It's currently set to `false` in library, I don't think we want to be publishing security results automatically.

File: config/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
```
Comment (AlvaroBrey) [inline config/workflows/scorecard.yml:66]:
```suggestion
```
Same as above, not sure if we want to publish the results as artifacts, as that would be publicly visible

**Review by AlvaroBrey — CHANGES_REQUESTED** [review]

**Comment by AlvaroBrey** [comment]
> If so, we would need to change sync.yml and also add this repo?

We could, but it can't just be added to the list of the other repos, as those currently sync all workflows in `config/workflows`, and this repo doesn't need all of the android stuff.

**Comment by AlvaroBrey** [comment]
> > If so, we would need to change sync.yml and also add this repo?
> 
> We could, but it can't just be added to the list of the other repos, as those currently sync all workflows in `config/workflows`, and this repo doesn't need all of the android stuff. Either way it can be done, if you want to.

Done at https://github.com/nextcloud/android-config/pull/47/commits/1c4fcbd768aef10664c7fc5ea6c4fc7772d13002

**Review by AlvaroBrey — APPROVED** [review]

**Comment by AlvaroBrey** [comment]
When this is synced to `android-library` we'll need to remove the old workflow, as it is called `scorecards.yml` instead of `scorecard.yml`

**Comment by tobiasKaminsky** [comment]
Nice catch!
https://github.com/nextcloud/android-library/pull/1066

**Pull request merged**